### ISSUE TITLE
ES6 Module & custom element support

### ISSUE THREAD

**Issue by kkimdev** [ajaxorg/ace#3682]
It will be great to have es6 module & custom element support.

```html
<script type="module" src="src/ace-module.js"></script>
<ace-editor></ace-editor>
```

**Comment by rjsteinert** [comment]
The closest thing -> https://github.com/Juicy/juicy-ace-editor/tree/ES6-modules

... but it appears this does not work well with the mode system which I bet would need to be refactored to support ES6 modules.

**Comment by whazor** [comment]
#5070 just got merged, which should help with the ES Modules, and is being released under Ace v1.15.3.

Please let us know whether it improves the situation for juicy-ace-editor.
